### ISSUE TITLE
Squeezed video problem after loading

### ISSUE THREAD

**Issue by Osjack** [afollestad/easy-video-player#25]
- Easy Video Player version: `0.2.9`
 - Device OS version: `Android 4.4.2`
 - Device Manufacturer: `Samsung`
 - Device Name: `Galaxy Note 8.0 (N5100)`
 
Xml layout
```java
<?xml version="1.0" encoding="utf-8"?>
<com.afollestad.easyvideoplayer.EasyVideoPlayer xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:id="@+id/player"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    app:evp_autoFullscreen="false"
    app:evp_autoPlay="true"
    app:evp_customLabelText="Hello!"
    app:evp_disableControls="false"
    app:evp_hideControlsOnPlay="true"
    app:evp_leftAction="restart"
    app:evp_pauseDrawable="@drawable/evp_action_pause"
    app:evp_playDrawable="@drawable/evp_action_play"
    app:evp_restartDrawable="@drawable/evp_action_restart"
    app:evp_retryText="@string/evp_retry"
    app:evp_submitText="@string/evp_submit" />
```

###### Reproduction Steps
1. Open player in vertical orientation 
2. While video loading rotate the device to horizontal orientation

###### Expected Result
Video loads and plays in fullscreen

###### Actual Result
Video loads and plays in half screen (squeezed)
![screenshot_2016-07-25-16-02-33](https://cloud.githubusercontent.com/assets/3192624/17103030/e8be2196-5284-11e6-876d-ec5ff9461e6e.png)

**Comment by CHPraxis** [comment]
More of a hack than a solution, but as a workaround, I added this to my activity's onCreate:

`//block rotation initially, until video loads. 
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_NOSENSOR);`

then:
`
@Override
    public void onStarted(EasyVideoPlayer player) {
        //resume rotation 
        //setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_SENSOR);

        Timer timer = new Timer();
        TimerTask task = new TimerTask() {
            @Override
            public void run() {
                EVActivity.this.runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_SENSOR);
                    }
                });
            }
        };
        timer.schedule(task,400);
    }`


Basically, it blocks rotation until 400 ms (0.4 seconds) after the video starts playing.  (My video autoplays upon download.)

I imagine you could probably put this in OnPrepared as an alternative if you don't autoplay your video.

This resolved the issue for me, with the only negative being the loss of a fraction of a second before rotation happens.

**Comment by juliancorrea** [comment]
I have the same problem. 
@afollestad, have a expected date for solve this BUG?

**Comment by afollestad** [comment]
@juliancorrea no, I can't reproduce it so I can't fix it.

**Comment by juliancorrea** [comment]
@CHPraxis, your solution/hack don't works for me.  
I found another way (more more of a hack than a solution) to solve.
Briefly, it was necessary to invoke the method responsible for updating the UI. As there is no exact place to do this, onBuffering was where I succeeded.
Here is the code:

    public class MyActivity extends AppCompatActivity implements EasyVideoCallback {
       private int initialWidth = 0;
       private int initialHeight = 0;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //your code ....
        initialHeight = getDisplayMetrics().heightPixels;
        initialWidth = getDisplayMetrics().widthPixels;
    }
    public void onBuffering(int percent) {
       //The condition "percent > 5"  is arbitrary. This condition is arbitrary, is an attempt to measure 
       //the correct time to start the video without resizing visually on the user's face.
       //For devices with very fast processors (quad-core, etc.), it will not work.
       fixRotationBugEasyVideoPlayer(percent > 5);
    }

    public void fixRotationBugEasyVideoPlayer(boolean percentFlag){
        DisplayMetrics dm = getDisplayMetrics();

        if(dm.widthPixels != initialWidth || dm.heightPixels != initialHeight)
            player.onSurfaceTextureSizeChanged(null, dm.widthPixels, dm.heightPixels);

        if(!player.isPlaying() && percentFlag) {
            player.start();
            player.hideControls();
        }
    }

        private DisplayMetrics getDisplayMetrics(){
            DisplayMetrics dm = new DisplayMetrics();
            getWindowManager().getDefaultDisplay().getMetrics(dm);
            return dm;
        }
    }
